### PR TITLE
Update Imagick.php

### PR DESCRIPTION
**Pull request by djbodya**
Fixing problem "Imagick::compositeImage(): Argument #3 ($x) must be of type int, float given". It appears when creating thumbnail template with "Add Overlay (Imagick)" and "Origin: center"

  
  

## Changes in this pull request  
Resolves #

### <samp>🤖 Generated by Copilot at 67ba2b9</samp>

Fixed a bug in the `Imagick` image adapter that caused an exception when overlaying images with float coordinates. Cast the coordinates to integers in the `addOverlay` function of `lib/Image/Adapter/Imagick.php`.

### <samp>🤖 Generated by Copilot at 67ba2b9</samp>

> _`addOverlay` fixed_
> _cast floats to integers now_
> _autumn bug no more_

### <samp>🤖 Generated by Copilot at 67ba2b9</samp>

*  Cast `$x` and `$y` parameters to integers in `addOverlay` function to fix exception bug ([link](https://github.com/pimcore/pimcore/pull/15632/files?diff=unified&w=0#diff-aa0869d1a589a6e48391adddef6c4e08b08dfb5c3943d560d758b7dd38bc1518L755-R755))

### CODE REVIEW

File: lib/Image/Adapter/Imagick.php [hunk 1]
```diff
@@ -752,7 +752,7 @@ public function addOverlay(mixed $image, int $x = 0, int $y = 0, int $alpha = 10
             }
 
             $newImage->evaluateImage(\Imagick::EVALUATE_MULTIPLY, $alpha, \Imagick::CHANNEL_ALPHA);
-            $this->resource->compositeImage($newImage, constant('Imagick::' . $composite), $x, $y);
+            $this->resource->compositeImage($newImage, constant('Imagick::' . $composite), intval($x), intval($y));
```
Comment (jdreesen) [inline lib/Image/Adapter/Imagick.php]: I think we should just cast:
```suggestion
            $this->resource->compositeImage($newImage, constant('Imagick::' . $composite), (int)$x, (int)$y);
```

**Comment by blankse** [comment]
Should be changed in branch `10.6`

**Comment by djbodya** [comment]
> Should be changed in branch `10.6`

I faced this issue in 11.0.5. Will the bug be fixed in this branch as well?

**Comment by blankse** [comment]
Yes, the branch `10.6` will merged to `11.0` and `11.x`. This should be fixed in all supported branches.

**Comment by dvesh3** [comment]
@djbodya please rebase on 10.6 branch as Sebastian pointed out. thanks!

**Comment by dvesh3** [comment]
Hi @djbodya changing the base branch here is not enough. It needs to be rebased properly on 10.6 branch. could you please take care of it? thanks!

**Comment by djbodya** [comment]
> Hi @djbodya changing the base branch here is not enough. It needs to be rebased properly on 10.6 branch. could you please take care of it? thanks!

Unfortunately I do not know how to do it properly. I was forked only one branch 11.x. Now I can not see possibility to rebase in my fork.

**Comment by kingjia90** [comment]
Closing in favor of https://github.com/pimcore/pimcore/pull/15738 which is rebased on 10.6

**Pull request closed without merging**